src/term.rs

```diff
@@ -826,6 +826,11 @@ impl fmt::Display for NAryOp {
     }
 }
 
+pub enum TraverseMethod {
+    TopToBottom,
+    BottomToTop,
```

Comment:
```suggestion
    BottomUp,
```

It's a nitpick, and it can actually sounds less coherent than the original version, but those are the accepted terms.